rest/taskana-rest-spring/src/test/java/pro/taskana/rest/TaskCommentControllerIntTest.java

```diff
@@ -0,0 +1,429 @@
+        };
+    assertThatThrownBy(httpCall)
+        .extracting(ex -> ((HttpClientErrorException) ex).getStatusCode())
+        .isEqualTo(HttpStatus.FORBIDDEN);
+  }
+
+  @Disabled("Disabled until Authorization check is up!")
+  @Test
+  void testCreateTaskCommentForNotVisibleTaskShouldFail() {
+
+    TaskCommentResource taskCommentResourceToCreate = new TaskCommentResource();
+    taskCommentResourceToCreate.setTaskId("TKI:000000000000000000000000000000000004");
+    taskCommentResourceToCreate.setTextField("newly created task comment");
+
+    ThrowingCallable httpCall =
+        () -> {
+          template.exchange(
+              restHelper.toUrl(
+                  Mapping.URL_TASK_COMMENTS, "TKI:000000000000000000000000000000000004"),
+              HttpMethod.POST,
+              new HttpEntity<>(taskCommentResourceToCreate, restHelper.getHeadersUser_1_1()),
+              ParameterizedTypeReference.forType(TaskCommentResource.class));
+        };
+    assertThatThrownBy(httpCall)
+        .extracting(ex -> ((HttpClientErrorException) ex).getStatusCode())
+        .isEqualTo(HttpStatus.FORBIDDEN);
+  }
+
+  @Test
+  void testCreateTaskCommentForNotExistingTaskShouldFail() {
+
+    TaskCommentResource taskCommentResourceToCreate = new TaskCommentResource();
+    taskCommentResourceToCreate.setTaskId("DefinatelyNotExistingId");
+    taskCommentResourceToCreate.setTextField("newly created task comment");
+
+    ThrowingCallable httpCall =
+        () -> {
+          template.exchange(
+              restHelper.toUrl(Mapping.URL_TASK_COMMENTS, "DefinatelyNotExistingId"),
+              HttpMethod.POST,
+              new HttpEntity<>(taskCommentResourceToCreate, restHelper.getHeadersAdmin()),
+              ParameterizedTypeReference.forType(TaskCommentResource.class));
+        };
+    assertThatThrownBy(httpCall)
+        .extracting(ex -> ((HttpClientErrorException) ex).getStatusCode())
+        .isEqualTo(HttpStatus.NOT_FOUND);
+  }
+
+  @Test
+  void testCreateTaskCommentWithDifferentTaskIdInResourceShouldFail() {
+
+    TaskCommentResource taskCommentResourceToCreate = new TaskCommentResource();
+    taskCommentResourceToCreate.setTaskId("TKI:000000000000000000000000000000000000");
+    taskCommentResourceToCreate.setTextField("newly created task comment");
+
+    ThrowingCallable httpCall =
+        () -> {
+          template.exchange(
+              restHelper.toUrl(Mapping.URL_TASK_COMMENTS, "DifferentTaskId"),
+              HttpMethod.POST,
+              new HttpEntity<>(taskCommentResourceToCreate, restHelper.getHeadersAdmin()),
+              ParameterizedTypeReference.forType(TaskCommentResource.class));
+        };
+    assertThatThrownBy(httpCall)
+        .extracting(ex -> ((HttpClientErrorException) ex).getStatusCode())
+        .isEqualTo(HttpStatus.BAD_REQUEST);
+  }
+
+  @Test
+  void testUpdateTaskCommentWithConcurrentModificationShouldFail() {
```

Comment:
Does it matter that this scenario is not 100% correct?
The "true" scenario is that someone else modified the comment and you tried to update an old comment. Not a newer one..

Since we are comparing the modified timestamps as of now this should't matter, right?

Reply:
Actually the test is doing exactely that. In the test there is an update of a task comment. (which then gets a more recent modified timestamp). Then there is an attempt to update that task comment again with a modified timestamp before the most recent update. 

So someone else modified it before the second update attempt which is resulting in an expected concurrency exception.